src/main/java/bosco/Bosco.java

```diff
@@ -0,0 +1,295 @@
+        selectedTask.markAsDone();
+        printMessages(MESSAGE_MARK_DONE, INDENT_EXTRA + selectedTask);
+    }
+
+    private static void executeUnmarkTask(String commandArgs) {
+        Task selectedTask = getSelectedTaskFromCommandArgs(commandArgs);
+        selectedTask.markAsNotDone();
+        printMessages(MESSAGE_MARK_UNDONE, INDENT_EXTRA + selectedTask);
+    }
+
+    private static void executeDeleteTask(String commandArgs) {
+        Task selectedTask = getSelectedTaskFromCommandArgs(commandArgs);
+        tasksList.remove(selectedTask);
+        printMessages(MESSAGE_DELETED_TASK, INDENT_EXTRA + selectedTask, getTaskCountMessage());
+    }
+
+    private static Task getSelectedTaskFromCommandArgs(String commandArgs) {
+        int taskNumber = Integer.parseInt(commandArgs);
+        if (taskNumber < 1 || taskNumber > tasksList.size()) {
+            throw new IndexOutOfBoundsException();
+        }
+        return tasksList.get(taskNumber - 1);
+    }
+
+    private static void executeAddTodo(String commandArgs) throws EmptyDescriptionException {
+        String description = commandArgs.strip();
+        if (description.isEmpty()) {
+            throw new EmptyDescriptionException();
+        }
+        addToTasksList(new Todo(description, false));
+    }
+
+    private static void executeAddDeadline(String commandArgs)
+            throws EmptyDescriptionException, MissingPrefixException {
+        int indexOfByPrefix = commandArgs.indexOf(DEADLINE_PREFIX_BY);
+        if (indexOfByPrefix == -1) {
+            throw new MissingPrefixException(DEADLINE_PREFIX_BY);
+        }
+        String description = commandArgs.substring(0, indexOfByPrefix).strip();
+        if (description.isEmpty()) {
+            throw new EmptyDescriptionException();
+        }
+        String by = removePrefix(commandArgs.substring(indexOfByPrefix), DEADLINE_PREFIX_BY).strip();
+        addToTasksList(new Deadline(description, false, by));
+    }
+
+    private static void executeAddEvent(String commandArgs)
+            throws EmptyDescriptionException, MissingPrefixException {
+        int indexOfFromPrefix = commandArgs.indexOf(EVENT_PREFIX_FROM);
+        if (indexOfFromPrefix == -1) {
+            throw new MissingPrefixException(EVENT_PREFIX_FROM);
+        }
+        int indexOfToPrefix = commandArgs.indexOf(EVENT_PREFIX_TO);
+        if (indexOfToPrefix == -1) {
+            throw new MissingPrefixException(EVENT_PREFIX_TO);
+        }
+        String description = commandArgs.substring(0, indexOfFromPrefix).strip();
+        if (description.isEmpty()) {
+            throw new EmptyDescriptionException();
+        }
+        String from = removePrefix(commandArgs.substring(indexOfFromPrefix, indexOfToPrefix),
+                EVENT_PREFIX_FROM).strip();
+        String to = removePrefix(commandArgs.substring(indexOfToPrefix), EVENT_PREFIX_TO).strip();
+        addToTasksList(new Event(description, false, from, to));
+    }
```

Comment:
Consider using a single method to parse the strings to reduce duplicated lines of code.